.github/workflows/scorecards-analysis.yml

```diff
@@ -0,0 +1,44 @@
+name: Scorecards supply-chain security
+on:
+  branch_protection_rule: {}
+  schedule:
+    - cron: '21 20 * * 3'
+  push:
+    branches: [ main ]
+
+permissions: read-all
+
+jobs:
+  analysis:
+    name: Scorecards analysis
+    runs-on: ubuntu-latest
+    permissions:
+      security-events: write
+      actions: read
+      contents: read
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+        with:
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@c8416b0b2bf627c349ca92fc8e3de51a64b005cf 
+        with:
+          results_file: results.sarif
+          results_format: sarif
+          repo_token: ${{ secrets.SCORECARD_READ_TOKEN }}
```

Comment:
does this need adding?

Reply:
no I have admin permissions on this repo so I could add it